docs/source/notebooks_and_ipython/kedro_and_notebooks.md

```diff
@@ -209,27 +221,86 @@ You don't need to restart the kernel for the `catalog`, `context`, `pipelines` a
+
+<details>
+<summary>Click to see an example.</summary>
+
+![jupyter_ipython_load_node](../meta/images/jupyter_ipython_load_node.gif)
+
+</details>
+
+---
+
+To be able to access your node's inputs, make sure they are explicitly defined in your project's catalog.
+
+You can then run the generated cells to recreate how the node would run in your pipeline. You can use this to explore your node's inputs, behaviour, and outputs in isolation, or for [debugging](#debugging-a-kedro-project-within-a-notebook).
+
+### `%run_viz` line magic
+
+``` {note}
+If you have not yet installed [Kedro-Viz](https://github.com/kedro-org/kedro-viz) for the project, run `pip install kedro-viz` in your terminal from within the project directory.
+```
+
+You can display an interactive visualisation of your pipeline directly in your notebook using the `%run_viz` line magic from within a cell:
+
+```python
+%run_viz
+```
+
+![View your project's Kedro Viz inside a notebook](../meta/images/run_viz_in_notebook.png)
 
 ## Debugging a Kedro project within a notebook
 
- You can use the `%debug` [line magic](https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-debug) to launch an interactive debugger in your Jupyter notebook. Declare it before a single-line statement to step through the execution in debug mode. You can use the argument `--breakpoint` or `-b` to provide a breakpoint.
-The follow sequence occurs when `%debug` runs immediately after an error occurs:
+ You can use the built-in [`%debug` line magic](https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-debug) to launch an interactive debugger in your Jupyter notebook. Declare it before a single-line statement to step through the execution in debug mode. You can use the argument `--breakpoint` or `-b` to provide a breakpoint. Alternatively, use the command with no arguments after an error occurs to load the stack trace and begin debugging.
+
+ The follow sequence occurs when `%debug` runs after an error occurs:
  - The stack trace of the last unhandled exception loads.
  - The program stops at the point where the exception occurred.
  - An interactive shell where the user can navigate through the stack trace opens.
 
  You can then inspect the value of expressions and arguments, or add breakpoints to the code.
 
+Here is example debugging workflow after discovering a node in your pipeline is failing:
+1. Inspect the logs to find the name of the failing node. We can see below the problematic node is `split_data_node`.
+
 <details>
-<summary>Click to see an example.</summary>
+<summary>Click to the pipeline failure logs.</summary>
+
+![pipeline_error_logs](../meta/images/pipeline_error_logs.png)
+
+</details>
+
+2. In your notebook, run `%load_node <name-of-failing-node>` to load the contents of the problematic node with the [`%load_node` line magic](#kedro-line-magics).
```

Comment:
The formatting on the `%load_node` line magic subheading was causing issues with Sphinx when referenced. Instead, I've linked to the line magic section as a whole.